### ISSUE TITLE
Show a count of overdue tasks in the nav bar

### ISSUE THREAD

**Issue by garethbowen** [medic/medic-webapp#3943]
Just like we have unread counters for reports and messages, add an overdue tasks counter in a bubble in the nav bar.

![unread-reports](https://user-images.githubusercontent.com/2064938/31197847-e616a7c4-a9ae-11e7-82af-5e82fde0f428.png)

**Comment by garethbowen** [comment]
This didn't make it in to 2.14.0 - moving to 2.15.0

**Comment by amandacilek** [comment]
@garethbowen @sglangevin @diannakane 

Talked through these bubbles with Dianna last night and we had some thoughts:

In summary, bubbles should be there to direct our attention to things. After thinking on it some more we decided we might actually like to remove the bubble on the Reports page as there’s nothing there that _requires attention_. You can log into your newsfeed to see what’s up on Facebook as easily as you can open your Reports tab to see what’s recently happened. There’s no bubble on Facebook saying, “You have 25 unread newsfeed posts.” We also don’t want users to have to click through and “read” every single report in the list in order to get rid of the bubble, as that list can get long very quickly. One alternative to completely removing the bubble would be if it was just a red dot (no number) that went away once you visited the page. But we may or may not want the other pages to function in this way...

Messages feels like an important place for a bubble so we'd like to keep it on Messages. 

Tasks seems like a useful place for the bubble too, with the caveat that we only include "Overdue" and due "Today" tasks in the bubble. Dianna: "I want the tasks dot to tell me what things I need to do today, regardless of whether they are late or on-time. But not things that are due tomorrow." If I have two tasks in my list, one was due yesterday, one is due today, and one is due in two days, we'd like the bubble number to be "2". If I have no overdue tasks or tasks due today, and I only have future tasks in the list, then the bubble should not appear, as there is nothing requiring my attention today.

**Comment by amandacilek** [comment]
We may explore a slightly different visual treatment of the bubbles to be implemented after the nav redesign. This might potentially include removing the number counts from the bubbles and just having the red dot to generally indicate items requiring attention on this page. Again, not decided on, but a potential. Given that and LOE with this, would we like to wait to add anything to the Tasks page until the nav update?

**Comment by garethbowen** [comment]
@amandacilek It looks like there is some discussion about wider design issues. @sglangevin and I agreed to bump this to 3.1.0 to give more time to complete the design.

**Comment by amandacilek** [comment]
I do not believe there are any outstanding questions about this. Desired updates are: 
- [ ] Add a red count bubble to the Tasks page in the nav bar, including "Overdue" and “Due today” in the count 
- [ ] Remove the red count bubble from the Reports page. There’s nothing there that specifically requires attention (maybe excepting the mRDT workflow). We don’t want users to have to click through and “read” every single report in the list in order to get rid of the bubble, as that list can get long very quickly. 
- [ ] Don’t let count bubbles go higher than “99+”. If count goes beyond 99, display “99+” to avoid needing to display numbers of three or more digits in length.

**Comment by kennsippell** [comment]
I've moved amandacilek's second update into a separate issue for implementation reasons. This issue is now relevant only to the tasks bubble. https://github.com/medic/medic/issues/5985

**Comment by andrablaj** [comment]
@PhilipNgari I added the screenshot from the [non-public ticket](https://gitlab.echis.go.ke/echis-core/config-echis-2-0/-/issues/2974) to this issue's description. Please feel free to add more details here as you see fit. Thank you!
